src/federated.c

```diff
@@ -150,7 +155,7 @@ void FederatedInputConnection_ctor(FederatedInputConnection *self, Reactor *pare
   self->delay = delay;
   self->type = type;
   self->last_known_tag = NEVER_TAG;
-  self->safe_to_assume_absent = FOREVER;
+  self->safe_to_assume_absent = 0; // FIXME: This should be set by the user
```

Comment:
Note that I change the default here. By setting this to 0 we move from a dataflow federation to a "poor-mans-ptides" where the after-delays on the federated connections should be greater than the actual network delays